lib/booking_system.rb

```diff
@@ -0,0 +1,82 @@
+require_relative 'date_range'
+require_relative 'reservation'
+require 'pry'
+
+module Hotel
+  class BookingSystem
+    attr_reader :reservations, :rooms
+
+    VALID_ROOM_NUMBERS = (1..20)
+    ROOM_COST = 200
+    DISCOUNT_ROOM_RATE = 150
+
+    def initialize
+      @reservations = []
+      @rooms = VALID_ROOM_NUMBERS
+    end
+
+
+    ####
+    # room: an Integer for the room number
+    # check_in: a Date object for the check in day
+    # check_out: a Date object for the check out day
+    def book_room(room_number, check_in, check_out)
+      unless VALID_ROOM_NUMBERS.include?(room_number)
+        raise ArgumentError
+      end
+
+      date_range = Hotel::DateRange.new(check_in, check_out)
+
+      unless available_rooms(date_range).include?(room_number)
+        raise ArgumentError
+      end
+
+      new_reservation = Hotel::Reservation.new(room_number, ROOM_COST, date_range)
+
+      @reservations << new_reservation
+      return new_reservation
+    end
+
+    def reservations_on_date(date)
+      reservations = []
+      @reservations.each do |reservation|
+        if reservation.within_reservation?(date)
```

Comment:
This is a good example of loose coupling. Rather than having to pull out the start and end dates from the reservation and do a bunch of math, you call a method and let the `Reservation` take care of it. Good work.